### ISSUE TITLE
What is equivalent method for ImageDataBunch.from_folder(path, train='train',valid='test') in Fastai-v2

### ISSUE THREAD

**Issue by actcod01** [fastai/fastai#3213]
Hello Guys,

### I have a separate image directory for (train, test,val) data. 
Dataset link kaggle _: paultimothymooney/chest-xray-pneumonia_
____

In **Fastai-v1** , I used `ImageDataBunch.from_folder(path, train='train',valid='test')` for assigning train and test datasets.

But how can I use the same approach in **Fastai-v2** for assigning train and test data in **Datablock?**

Please help,

Thanks,

**Comment by KBodolai** [comment]
Hi!

you create a DataBlock and you pass a GrandparentSplitter, something like this:

```
    dblock = DataBlock(blocks = (ImageBlock, CategoryBlock),
                   get_items = get_image_files,
                   splitter = GrandparentSplitter(train_name = 'train', 
                                                  valid_name = 'val'),
                   get_y = parent_label)
```

GrandparentSplitter will get as train and validation sets the names of the folder you specify, and in this case, you'll get the labels from the parent folders (NORMAL or PNEUMONIA, i think).

**Comment by actcod01** [comment]
> Hi!
> 
> you create a DataBlock and you pass a GrandparentSplitter, something like this:
> 
> ```
>     dblock = DataBlock(blocks = (ImageBlock, CategoryBlock),
>                    get_items = get_image_files,
>                    splitter = GrandparentSplitter(train_name = 'train', 
>                                                   valid_name = 'val'),
>                    get_y = parent_label)
> ```
> 
> GrandparentSplitter will get as train and validation sets the names of the folder you specify, and in this case, you'll get the labels from the parent folders (NORMAL or PNEUMONIA, i think).
> 
> then you create the dataloader from that datablock: `dls = dblock.dataloaders(data)`

**Fastai-v2.2.5**
Thanks a lot :+1:  It works for me.
